packages/api/endpoints/granules.js

```diff
@@ -940,6 +949,139 @@ async function del(req, res) {
+ * ('error' or 'skip').
+ * @param {number} [req.body.s3MultipartChunkSizeMb] - The S3 multipart chunk size in MB
+ * @param {string} [req.body.executionName] - Override to allow specifying an execution 'name'
+ * @param {object} req.testContext - The test context object
+ * @param {object} res - The response object.
+ * @returns {Promise<Object>} The response object with the execution ARN and message.
+ */
+async function bulkChangeCollection(req, res) {
+  const {
+    knex = await getKnexClient(),
+    sfnMethod = sfn,
+    workflow = 'MoveGranuleCollectionsWorkflow',
+  } = req.testContext || {};
+
+  const collectionPgModel = new CollectionPgModel();
+  const granulePgModel = new GranulePgModel();
+
+  if (!process.env.system_bucket) {
+    return res.boom.badRequest('API is misconfigured, system_bucket must be defined in the env variables');
+  }
+  if (!process.env.stackName) {
+    return res.boom.badRequest('API is misconfigured, stackName must be defined in the env variables');
+  }
+
+  const body = parsebulkChangeCollectionPayload(req.body);
+  if (isError(body)) {
+    return returnCustomValidationErrors(res, body);
+  }
+
+  //get collection
+  const pgCollection = await collectionPgModel.get(
+    knex,
+    deconstructCollectionId(body.sourceCollectionId)
+  );
+  const query = granulePgModel.queryBuilderSearch(knex, {
+    collection_cumulus_id: pgCollection.cumulus_id,
+  });
+  query.select('granule_id');
+  query.limit(body.batchSize);
+  const granules = await query;
+  if (granules.length === 0) {
+    return res.boom.notFound(
+      `No granules found for collection ${body.sourceCollectionId}`
+    );
+  }
+
+  const { name, version } = deconstructCollectionId(body.sourceCollectionId);
+  const executionName = body.executionName || uuidv4();
+
+  let stateMachineArn;
+  try {
+    const workflowArnObject = await getJsonS3Object(
+      process.env.system_bucket,
+      getWorkflowFileKey(process.env.stackName, workflow)
+    );
+    stateMachineArn = workflowArnObject.arn;
+  } catch (error) {
+    return res.boom.badRequest(
+      `Unable to find state machine ARN for workflow ${workflow}`
+    );
+  }
+  const input = await buildPayload({
+    workflow,
+    cumulus_meta: {
+      workflow_start_time: Date.now(),
+      execution_name: executionName,
+      state_machine: stateMachineArn,
+      system_bucket: process.env.system_bucket,
+    },
+    meta: {
+      collection: {
+        name,
+        version,
+      },
+      bulkChangeCollection: {
+        batchSize: body.batchSize,
+        cmrGranuleUrlType: body.cmrGranuleUrlType,
+        concurrency: body.concurrency,
+        invalidBehavior: body.invalidBehavior,
+        s3MultipartChunkSizeMb: body.s3MultipartChunkSizeMb,
+        targetCollection: deconstructCollectionId(body.targetCollectionId),
+      },
+    },
+    payload: {
+      granuleIds: granules.map((granule) => granule.granule_id),
```

Comment:
This is going to be limited to something like 32KB.   We're going to need to write it to S3 a'la CMA.

Reply:
Done: df11eae9e82

Reply:
LGTM